plugins/woocommerce-admin/client/marketing/overview-multichannel/DiscoverTools/useRecommendedPlugins.ts

```diff
@@ -1,29 +1,42 @@
 /**
  * External dependencies
  */
-import { useSelect } from '@wordpress/data';
+import { useCallback } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import { STORE_KEY } from '~/marketing/data/constants';
 import { Plugin } from './types';
 
+const selector = 'getRecommendedPlugins';
 const category = 'marketing';
 
 type SelectResult = {
 	isLoading: boolean;
 	plugins: Plugin[];
+	removeRecommendedPlugin: ( slug: string ) => void;
 };
 
 export const useRecommendedPlugins = () => {
+	const { removeRecommendedPlugin } = useDispatch( STORE_KEY );
+
+	const callback = useCallback(
```

Comment:
Not saying this is wrong, just curious for the reasons behind wrapping in `useCallback`.  Does `removeRecommendedPlugin` change during the course of render?

Reply:
When I was developing it, without the `useCallback`, there were infinite renders and network requests. However, I just tried it again and I couldn't reproduce it anymore 🤷 😅  . I have removed it in [`ffbe20e` (#35542)](https://github.com/woocommerce/woocommerce/pull/35542/commits/ffbe20e4cacc33cbf6fb33f9979da2ffdfc76e21).